core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -432,11 +443,22 @@ public void run(Throwable cause) {
                 // This method will be invoked only when `canSchedule()` returns true.
                 assert state != State.DONE;
 
-                failAndRespond(cause, serviceUnavailableResponse, Http2Error.INTERNAL_ERROR, true);
+                failAndRespond(cause, convertException(cause), Http2Error.INTERNAL_ERROR, true);
             }
         };
     }
 
+    private AggregatedHttpResponse convertException(Throwable cause) {
+        final AggregatedHttpResponse convertedResponse =
+                reqCtx.config().server().config().exceptionHandler().convert(reqCtx, cause);
```

Comment:
Could we have per-virtualhost and per-service exception handler similarly, probably in a follow-up PR for 1.7?

Reply:
I thought it was too much but yeah we need that probably. 😄 
Let me make a follow-up PR for that. Thanks!